src/containers/scrollable-canvas.jsx

```diff
@@ -0,0 +1,174 @@
+            }
+            nextProps.canvas.addEventListener('wheel', this.handleWheel);
+        }
+    }
+    handleHorizontalScrollbarMouseDown (event) {
+        window.addEventListener('mousemove', this.handleHorizontalScrollbarMouseMove);
+        window.addEventListener('mouseup', this.handleHorizontalScrollbarMouseUp);
+        this.initialMouseX = getEventXY(event).x;
+        this.initialScreenX = paper.view.matrix.tx;
+        event.preventDefault();
+    }
+    handleHorizontalScrollbarMouseMove (event) {
+        const dx = this.initialMouseX - getEventXY(event).x;
+        paper.view.matrix.tx = this.initialScreenX + (dx * paper.view.zoom * 2);
+        clampViewBounds();
+        this.props.updateViewBounds(paper.view.matrix);
+        event.preventDefault();
+    }
+    handleHorizontalScrollbarMouseUp () {
+        this.initialMouseX = null;
+        this.initialScreenX = null;
+        window.removeEventListener('mousemove', this.handleHorizontalScrollbarMouseMove);
+        window.removeEventListener('mouseup', this.handleHorizontalScrollbarMouseUp);
+        event.preventDefault();
+    }
+    handleVerticalScrollbarMouseDown (event) {
+        window.addEventListener('mousemove', this.handleVerticalScrollbarMouseMove);
+        window.addEventListener('mouseup', this.handleVerticalScrollbarMouseUp);
+        this.initialMouseY = getEventXY(event).y;
+        this.initialScreenY = paper.view.matrix.ty;
+        event.preventDefault();
+    }
+    handleVerticalScrollbarMouseMove (event) {
+        const dy = this.initialMouseY - getEventXY(event).y;
+        paper.view.matrix.ty = this.initialScreenY + (dy * paper.view.zoom * 2);
+        clampViewBounds();
+        this.props.updateViewBounds(paper.view.matrix);
+        event.preventDefault();
+    }
+    handleVerticalScrollbarMouseUp (event) {
+        this.initialMouseY = null;
+        this.initialScreenY = null;
+        window.removeEventListener('mousemove', this.handleVerticalScrollbarMouseMove);
+        window.removeEventListener('mouseup', this.handleVerticalScrollbarMouseUp);
+        event.preventDefault();
+    }
+    handleWheel (event) {
+        // Multiplier variable, so that non-pixel-deltaModes are supported. Needed for Firefox.
+        // See #529 (or LLK/scratch-blocks#1190).
+        const multiplier = event.deltaMode === 0x1 ? 15 : 1;
+        const deltaX = event.deltaX * multiplier;
+        const deltaY = event.deltaY * multiplier;
+        if (event.metaKey || event.ctrlKey) {
+            // Zoom keeping mouse location fixed
+            const canvasRect = this.props.canvas.getBoundingClientRect();
+            const offsetX = event.clientX - canvasRect.left;
+            const offsetY = event.clientY - canvasRect.top;
+            const fixedPoint = paper.view.viewToProject(
+                new paper.Point(offsetX, offsetY)
+            );
+            zoomOnFixedPoint(-deltaY / 1000, fixedPoint);
```

Comment:
That denominator used to be `100` is the `1000` a typo or intentional?

Reply:
Yes, I felt like scroll zoom was way too fast, so I took it down a factor of 10! Sorry, I tried to sneak that in.